tmt/steps/provision/testcloud.py

```diff
@@ -972,6 +983,22 @@ def start(self) -> None:
 
         self._apply_hw_arch(self._domain, self.is_kvm, self.is_legacy_os)
 
+        # Do we have virtiofs to handle?
+        if self.virtiofs:
+            if (f"qemu:///{self.connection}" == "qemu:///session" and
+                    libvirt.getVersion() < 10000000):
+                raise tmt.utils.ProvisionError("Only system connection is supported"
+                                               " for virtiofs on libvirt < 10.0.")
+
+            virtiofs_split = self.virtiofs.split(":")
+            if len(virtiofs_split) != 2:
+                raise tmt.utils.ProvisionError("virtiofs specified in a bad format, use "
+                                               "<host path>:<guest path>")
+
+            self._domain.virtiofs_configuration = [
+                VIRTIOFSConfiguration(
```

Comment:
Need to check if the <host path> and <guest path> are absolute paths. Relative host path is invalid to testcloud or libvirt.  Relative guest path doesn't make sense to guest.

Reply:
Good point, what'd be the best way to tackle? Just take a look if host path starts with '/', and prepend current dir if it doesn't?

Reply:
`os.path` module will help that checking.
For the host directory, I think user should prepare them. For the guest directory, because guest is provisioned, so testcloud or tmt should prepare it. The host path could be a file or non-exist, for the former, tmt can pre-create it and record that in log; for the latter, just raise an error.   
So the code will be like:
```py
if not os.path.isabs(host_path):
    raise ProvisionError(f"Cannot apply hardware requirement '{constraint}', the host path is not absolute.")
if os.path.isfile(host_path):
    raise ProvisionError(f"Cannot apply hardware requirement '{constraint}', the host path is an existing file.")
if not os.path.exists(host_path):
    os.mkdir(host_path)
    logger.debug(f"the host path {constraint.value} don't exist, create it by tmt ")
```
For the guest path, I think such checking will help. Can we check the 2nd and the 3rd conditions for guest by tmt? Otherwise, we can check them in testcloud.  
  
Now I test testcloud for the non-exist guest path and guest file path.   
For the non-exist guest path, testcloud will create it and mount the virtiofs to it. It is expected
version: testcloud-0.9.13-1.fc41.noarch
```
➜  ~ testcloud create fedora:40 --virtiofs /tmp/export:/mnts

[cloud-user@testcloud ~]$ mount|grep virtiofs
virtiofs-0 on /mnts type virtiofs (rw,relatime)
```
For the file path, the virtiofs will not mounted. And it causes the cloud-final.service failed:
```
➜  ~ ssh -oStrictHostKeyChecking=no -oUserKnownHostsFile=/dev/null cloud-user@192.168.124.215
Warning: Permanently added '192.168.124.215' (ED25519) to the list of known hosts.
cloud-user@192.168.124.215's password: 
[systemd]
Failed Units: 1
  cloud-final.service

[cloud-user@testcloud ~]$ mount|grep virtiofs
```
I think it is better to check the file path first and raise an error.

Reply:
Unrelated to the proposal, just one note: please, use `pathlib` in tmt code, `os.path` interface is considered legacy and should not be used.